syllabus/core/stat_recorder.py

```diff
@@ -1,35 +1,39 @@
+import os
+import json
 import warnings
 import numpy as np
 from syllabus.task_space import TaskSpace
 from typing import Callable
 from gymnasium.spaces import Discrete #, MultiDiscrete?
-import json
-import os
+from collections import deque, defaultdict
 
 class StatRecorder:
     """
     Individual stat tracking for each task.
     """
 
-    def __init__(self, task_space: TaskSpace, task_names: Callable = None, calc_past_N=None):
+    def __init__(self, task_space: TaskSpace, keep_last_N=10, calc_past_N=None):
         """Initialize the StatRecorder"""
 
         self.task_space = task_space
-        self.task_names = task_names
         self.calc_past_N = calc_past_N
 
         assert isinstance(self.task_space, TaskSpace), f"task_space must be a TaskSpace object. Got {type(task_space)} instead."
         assert isinstance(self.task_space.gym_space, Discrete), f"Only Discrete task spaces are supported. Got {type(task_space.gym_space)}"
+        if calc_past_N and calc_past_N > keep_last_N:
```

Comment:
```suggestion
        if calc_past_N is not None and calc_past_N > keep_last_N:
```

Reply:
These should just be one option. So if calc_past_N is None, store nothing aside from the running means. If calc_past_n is not None, store that many values, and calculate using all of the stored values.